[sitecore-jss-nextjs] Personalize middleware: geolocation data can now be passed to personalize middleware from application level (#2084)
Sitecore/jss: CHANGELOG.md

### 🛠 Breaking Changes

> * `[Next.js]` `[React]`: `[sitecore-jss]` `[sitecore-jss-react]` `[sitecore-jss-nextjs]` `[create-sitecore-jss]` Upgrade React to version 19 and Nextjs to version 15 ([#2078](https://github.com/Sitecore/jss/pull/2078)) ([#2084](https://github.com/Sitecore/jss/pull/2084)):

Reviewer: ```suggestion
* `[Next.js]` `[React]`: `[sitecore-jss]` `[sitecore-jss-react]` `[sitecore-jss-nextjs]` `[create-sitecore-jss]` Upgrade React to version 19 and Nextjs to version 15 ([#2078](https://github.com/Sitecore/jss/pull/2078))([#2084](https://github.com/Sitecore/jss/pull/2084)):
```

  * upgrade React and Nextjs dependencies for the new major versions
  * with React 19, JSX is in the 'react' namespace and therefore 'react' needs to be imported befoore using JSX. All OOTB react and nextjs components have been updated
  * `react-test-renderer` has been deprecated in react 19. additionaly `enzyme` is not supported anymore so all unit tests have been migrated to use `@`testing-library/react`
  * `propTypes` have been deprecated by react and have been removed from the solution

>   * in NextJs 15 the `geo` and `ip` properties on `NextRequest` have been removed. To account for this `sitecore-cloudsdk` dependencies have been upgraded to 0.5.1, which does not include breaking change.

Reviewer: ```suggestion
  * in NextJs 15 the `geo` and `ip` properties on `NextRequest` have been removed. To account for this `@sitecore-cloudsdk` dependencies have been upgraded to 0.5.1, which does not include breaking changes, however you will have to upgrade your cloudsdk dependencies to meet peer dependencies requirements
```

  * remove 'react' dependency from nextconfig webpack externals in monorepo next config plugin as it is not needed anymore.
  * PersonalizeMiddleware handler now accepts PersonalizeOptions, that can be used to provide geolocation data from application level

> * `[Angular]`: `[sitecore-jss-proxy]` `[sitecore-jss-angular]` - `sitecore-cloudsdk` dependencies have been upgraded to 0.5.1, which does not include breaking change. ([#2060](https://github.com/Sitecore/jss/pull/2060))([#2078](https://github.com/Sitecore/jss/pull/2078))

Reviewer: ```suggestion
* `[Angular]`: `[sitecore-jss-proxy]` `[sitecore-jss-angular]` `@sitecore-cloudsdk` dependencies have been upgraded to 0.5.1 ([#2060](https://github.com/Sitecore/jss/pull/2060))([#2078](https://github.com/Sitecore/jss/pull/2078))
```

* Upgrade cloudsdk to 0.5 ([#2060](https://github.com/Sitecore/jss/pull/2060)):
  * This upgrade doesn't introduce any breaking changes, however you will have to upgrade your cloudsdk dependencies to meet peer dependencies requirements

> * `[sitecore-jss]` `[sitecore-jss-react]` `[sitecore-jss-nextjs]` `[create-sitecore-jss]` `[sitecore-jss-proxy]` `[sitecore-jss-angular]` Upgrade React to version 19 and Nextjs to version 15 ([#2078](https://github.com/Sitecore/jss/pull/2078)) ([#2084](https://github.com/Sitecore/jss/pull/2084)):

Reviewer: ```suggestion
* `[sitecore-jss]` `[sitecore-jss-react]` `[sitecore-jss-nextjs]` `[create-sitecore-jss]` Upgrade React to version 19 and Nextjs to version 15 ([#2078](https://github.com/Sitecore/jss/pull/2078)) ([#2084](https://github.com/Sitecore/jss/pull/2084)):
```

  * upgrade React and Nextjs dependencies for the new major versions
  * with React 19, JSX is in the 'react' namespace and therefore 'react' needs to be imported befoore using JSX. All OOTB react and nextjs components have been updated
  * `react-test-renderer` has been deprecated in react 19. additionaly `enzyme` is not supported anymore so all unit tests have been migrated to use `@`testing-library/react`
  * `propTypes` have been deprecated by react and have been removed from the solution
  * in NextJs 15 the `geo` and `ip` properties on `NextRequest` have been removed. To account for this `sitecore-cloudsdk` dependencies have been upgraded to 0.5.1, which does not include breaking change.
  * remove 'react' dependency from nextconfig webpack externals in monorepo next config plugin as it is not needed anymore.

>   * for `[sitecore-jss-proxy]` `[sitecore-jss-angular]` - `sitecore-cloudsdk` dependencies have been upgraded to 0.5.1, which does not include breaking change.

Reviewer: Can it be moved under the different bullet point with the link to the 2078 PR?
